src/middleware/balance_capacity.rs

```diff
@@ -54,6 +57,11 @@ impl Handler for BalanceCapacity {
             super::log_request::add_custom_metadata(request, "in_flight_requests", count);
         }
 
+        // In report-only mode we serve all requests and only enforce the logging limit above
+        if self.report_only {
+            return handler.call(request);
+        }
```

Comment:
Instead of early exiting, it'd be nice to add some kind of log metadata like "would be dropped" to requests that would be dropped. This way we can analyze the impact with a single query in Papertrail.